src/test_application/activescenes/scenarios_physicstest.cpp

```diff
@@ -207,40 +186,60 @@ ActiveEnt add_quick_shape(
     SysHierarchy::add_child(rScene.m_basic.m_hierarchy, rScene.m_hierRoot, root);
 
     // Create collider / drawable to the root entity
-    ActiveEnt collider = add_solid(
+    add_solid(
             rScene, root, shape,
             Matrix4::scaling(size),
-            gc_mat_visualizer, 0.0f);
+            rScene.m_matVisualizer, 0.0f);
 
     // Make ball root a dynamic rigid body
-    rScene.m_physics.m_hasColliders.emplace(root);
-    rScene.m_physics.m_physBody.emplace(root);
-    rScene.m_physics.m_physLinearVel.emplace(root);
-    rScene.m_physics.m_physAngularVel.emplace(root);
-    osp::active::ACompPhysDynamic &rDyn = rScene.m_physics.m_physDynamic.emplace(root);
+    rScnPhys.m_physics.m_hasColliders.emplace(root);
+    rScnPhys.m_physics.m_physBody.emplace(root);
+    rScnPhys.m_physics.m_physLinearVel.emplace(root);
+    rScnPhys.m_physics.m_physAngularVel.emplace(root);
+    osp::active::ACompPhysDynamic &rDyn = rScnPhys.m_physics.m_physDynamic.emplace(root);
     rDyn.m_totalMass = 1.0f;
 
     // make gravity affect ball
-    rScene.m_hasGravity.emplace(root);
+    rScnTest.m_hasGravity.emplace(root);
 
     // Remove ball when it goes out of bounds
-    rScene.m_removeOutOfBounds.emplace(root);
+    rScnTest.m_removeOutOfBounds.emplace(root);
 
     // Set velocity
-    rScene.m_physIn.m_setVelocity.emplace_back(root, velocity);
+    rScnPhys.m_physIn.m_setVelocity.emplace_back(root, velocity);
 
     return root;
 }
 
-entt::any setup_scene(osp::Resources& rResources, osp::PkgId pkg)
+void PhysicsTest::setup_scene(CommonTestScene &rScene, osp::PkgId pkg)
 {
     using namespace osp::active;
 
-    entt::any sceneAny = entt::make_any<PhysicsTestScene>();
-    PhysicsTestScene &rScene = entt::any_cast<PhysicsTestScene&>(sceneAny);
+    // It should be clear that the physics test scene is a composition of:
+    // * PhysicsData:       Generic physics data
+    // * ACtxNwtWorld:      Newton Dynamics Physics engine
+    // * PhysicsTestData:   Additional scene-specific data, ie. dropping blocks
+    auto &rScnPhys  = rScene.emplace<PhysicsData>();
+    auto &rScnNwt   = rScene.emplace<ospnewton::ACtxNwtWorld>(gc_threadCount);
+    auto &rScnTest  = rScene.emplace<PhysicsTestData>();
+
+    // Add cleanup function to deal with reference counts
+    // Note: The problem with destructors, is that REQUIRE storing pointers in
```

Comment:
??

destructors for std::unique_ptr ?

if you make the destructor a stateless functor, you don't need to store anything extra.
```cpp
struct Bob
{
    void operator()(whatever)
    {
    }
}
std::unique_ptr<whatever, Bob> ...
```

Reply:
The problem here is that PhysicsTestData needs a way to access the CommonTestScene it is stored inside when destructed. I'm not sure how a unique ptr can be used to fix this.